instrumentation/pulsar/pulsar-2.8/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarMetricsRegistry.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.v2_8;
+
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.ATTRIBUTE_CONSUMER_NAME;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.ATTRIBUTE_PRODUCER_NAME;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.ATTRIBUTE_QUANTILE;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.ATTRIBUTE_RESPONSE_STATUS;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.ATTRIBUTE_SUBSCRIPTION;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.ATTRIBUTE_TOPIC;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.CONSUMER_METRICS_PREFIX;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.PRODUCER_METRICS_PREFIX;
+import static io.opentelemetry.javaagent.instrumentation.pulsar.v2_8.PulsarMetricsUtil.SCOPE_NAME;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.pulsar.client.impl.ConsumerImpl;
+import org.apache.pulsar.client.impl.ProducerImpl;
+
+public class PulsarMetricsRegistry {
+
+  private final Set<ProducerImpl<?>> producers = Collections.synchronizedSet(new HashSet<>());
+  private final Set<ConsumerImpl<?>> consumers = Collections.synchronizedSet(new HashSet<>());
+  private final Meter meter =
+      GlobalOpenTelemetry.getMeterProvider().meterBuilder(SCOPE_NAME).build();
+
+  public void init() {
```

Comment:
just move the init method's logic to Constructor?

Reply:
Because of the `PulsarMetricsRegistryDisabled` inherits `PulsarMetricsRegistry`. If the metrics is disabled, the metrics should be created.

Reply:
Oh, yes, I missed it